sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeCustomFormsOperation.cs

```diff
@@ -92,17 +96,22 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
                     ? await _serviceClient.GetAnalyzeFormResultAsync(new Guid(_modelId), new Guid(Id), cancellationToken).ConfigureAwait(false)
                     : _serviceClient.GetAnalyzeFormResult(new Guid(_modelId), new Guid(Id), cancellationToken);
 
-                // TODO: Handle correctly according to returned status code
-                // https://github.com/Azure/azure-sdk-for-net/issues/10386
                 // TODO: Add reasonable null checks.
 
-                if (update.Value.Status == OperationStatus.Succeeded || update.Value.Status == OperationStatus.Failed)
+                _response = update.GetRawResponse();
+
+                if (update.Value.Status == OperationStatus.Succeeded)
                 {
                     _hasCompleted = true;
                     _value = ConvertToRecognizedForms(update.Value.AnalyzeResult);
                 }
+                else if (update.Value.Status == OperationStatus.Failed)
+                {
+                    _hasCompleted = true;
+                    _value = new List<RecognizedForm>();
```

Comment:
This is where I was thinking would change to have `.Value` throw (and the getter for `Value`).  I'll file an issue for this for Preview 3, because it might take a little discussion to get the design right around this.  Azure.Core has a similar concept with `NoBodyResponse` which will throw if you try to access its value.  Technically, there is a response body here, but not the value you requested in the payload.  It will be good to bubble up this issue to the internal .NET channel for discussion.

Reply:
Technically we are already throwing when the `Value` property is called, but it has no value.

https://github.com/Azure/azure-sdk-for-net/blob/b97a3619b5b535efe544e8da3a9491ec8713032b/sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeCustomFormsOperation.cs#L33

But the error message wouldn't be accurate in this case:

```
"The operation has not completed yet."
```

We could copy the logic and update the getter. Do you think we should keep it as it is for Preview 1?

Reply:
#11466